processor/src/lib.rs

```diff
@@ -444,3 +445,19 @@ where
         )
     }
 }
+
+// INTERNALS
+// ================================================================================================
+
+#[cfg(feature = "internals")]
```

Comment:
I would change this to `#[cfg(any(test, feature = "internals"))]` (and update line 140 accordingly). This way, we don't need to add any extra attributes to internal test modules.